app/app/src/main/java/org/schabi/newpipe/crashreporttomarkdown/MainActivity.java

```diff
@@ -0,0 +1,179 @@
+package org.schabi.newpipe.crashreporttomarkdown;
+
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.app.AppCompatDelegate;
+import androidx.appcompat.widget.Toolbar;
+
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+public class MainActivity extends AppCompatActivity {
+
+    private static final String THEME = "theme";
+    private EditText jsonEditText;
+    private EditText humanReadableEditText;
+    private Button transform;
+    private Button copy;
+    private Button clear;
+    private Button preview;
+    private ClipboardManager clipboardManager;
+    private ClipData clipData;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        jsonEditText = findViewById(R.id.editTextError);
+        humanReadableEditText = findViewById(R.id.markdownEditText);
+        transform = findViewById(R.id.transform_button);
+        copy = findViewById(R.id.copy_button);
+        clear = findViewById(R.id.clear);
+        preview = findViewById(R.id.preview_button);
+        clipboardManager = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
+        checkTheme();
+        checkIntents();
+
+        //LISTENERS:
+        FloatingActionButton githubIssue = findViewById(R.id.github_button);
+        githubIssue.setOnClickListener(new View.OnClickListener() {
```

Comment:
Since you use many OnClickListeners, I suggest that you make MainActivity implement View.OnClickListener and implement one onClick function. Then you can distinguish views based on their ids

Reply:
You mean [like this](https://stackoverflow.com/a/30142323)?
Thanks for the tips